File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = {}
        n=0
        for i in nums:
            diff = target - i
            if diff in res:
                return [res[diff], n]
            res[i] = n
            nums[n]=None
            n+=1
        

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map1 = new Map();
    for (var i=0;i<nums.length;i++) {
        var diff=target-nums[i];
        if (map1[diff]!==undefined){
            return [map1[diff],i];
        }else{
            map1[nums[i]]=i;
        }
        
    }
    
    
};